### DISCUSSION TITLE
npx prisma db push error？

### DISCUSSION THREAD

**Discussion by lddlww** [prisma/prisma#18744] [Q&A] [answered]
```
npx prisma db push
> Downloading Prisma engines for Node-API for rhel-openssl-1.1.x [                    ] 0%Error: request to https://binaries.prisma.sh/all_commits/659ef412370fa3b41cd7bf6e94587c1dfb7f67e7/rhel-openssl-1.1.x/libquery_engine.so.node.gz.sha256 failed, reason: connect EINVAL 0.0.254.149:80 - Local (0.0.0.0:0)
```

how can i do?

**Comment by ludralph** [comment]
Hi @lddlww 

Can you please tell us what version of Prisma you are using?

**Reply by lddlww** [reply]
@ludralph prisma version is ```4.12.0```

i solved above issue following https://www.prisma.io/docs/reference/api-reference/environment-variables-reference#proxy-environment-variables

but i encountered another problem：
```
npx prisma db push
> Downloading Prisma engines for Node-API for rhel-openssl-1.1.x [                    ] 0%Error: request to https://binaries.prisma.sh/all_commits/659ef412370fa3b41cd7bf6e94587c1dfb7f67e7/rhel-openssl-1.1.x/migration-engine.gz.sha256 failed, reason: write EPROTO 139759518070656:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:../deps/openssl/openssl/ssl/record/ssl3_record.c:332
```

**Reply by ludralph** [reply]
Can you use this binaryTarget `rhel-openssl-1.0.x` instead of `rhel-openssl-1.1.x`?

**Reply by lddlww** [reply]
i tried, but it also throw error，or am i missing something？
.
```
sed -i 's/rhel-openssl-1.1.x/rhel-openssl-1.0.x/g' `grep -R "rhel-openssl-1.1.x" -rl *`

npx prisma db push
Error: Unknown binaryTarget rhel-openssl-1.1.x and no custom engine files were provided
```

**Reply by ludralph** [reply]
Hi @lddlww, Can you please specify how you set the binaryTarget and which operating system are you running on?

**Reply by lddlww** [reply]
centos8

**Reply by ludralph** [reply]
can you please share your schema file?

**Reply by ludralph** [reply]
Hey @lddlww ,

it would be really helpful if you could provide the additional information that we had asked about last time. If your problem is resolved by now or somehow doesn't feel relevant any more, you can simply close the Discussion as well.

**Reply by lddlww** [reply, marked as answer]
yes，it's always throw error，and i used follow steps to solve the issue

### 1、download component  by manual(yes, wget is ok)
```
wget  https://binaries.prisma.sh/all_commits/659ef412370fa3b41cd7bf6e94587c1dfb7f67e7/rhel-openssl-1.1.x/libquery_engine.so.gz
wget  https://binaries.prisma.sh/all_commits/659ef412370fa3b41cd7bf6e94587c1dfb7f67e7/rhel-openssl-1.1.x/migration-engine.gz
wget  https://binaries.prisma.sh/all_commits/659ef412370fa3b41cd7bf6e94587c1dfb7f67e7/rhel-openssl-1.1.x/prisma-fmt.gz
wget  https://binaries.prisma.sh/all_commits/659ef412370fa3b41cd7bf6e94587c1dfb7f67e7/rhel-openssl-1.1.x/query-engine.gz
```
### 2、set environment
```
vim ~./bash_profile
ENGINE_DIR="${WGET_DIR}"
export PRISMA_QUERY_ENGINE_LIBRARY="${WGET_DIR}/libquery_engine.so.node"
export PRISMA_QUERY_ENGINE_BINARY="${WGET_DIR}/query-engine"
export PRISMA_MIGRATION_ENGINE_BINARY="${WGET_DIR}/migration-engine"
export PRISMA_FMT_BINARY="${WGET_DIR}/prisma-fmt"
```

then, run ```npx prisma db push``` is ok now

**Reply by ludralph** [reply]
Thanks for sharing. It's really helpful

**Reply by Young-Jeff** [reply]
Note that PRISMA_MIGRATION_ENGINE_BINARY is deprecated in version 5.0.0 and should be replaced with PRISMA_SCHEMA_ENGINE_BINARY

**Reply by BG7ZAG** [reply]
Yes, I also noticed it yesterday. I made the following changes based on the comments above:
https://blog.bg7zag.com/2852
